pkg/controller/controller.go

```diff
@@ -173,18 +183,27 @@ func (c *Controller) runController() {
 
 		err = c.params.DoFunc()
 		if err != nil {
-			c.getLogger().WithField(fieldConsecutiveErrors, errorRetries).
-				WithError(err).Debug("Controller run failed")
-			c.recordError(err)
-
-			if !c.params.NoErrorRetry {
-				if c.params.ErrorRetryBaseDuration != time.Duration(0) {
-					interval = time.Duration(errorRetries) * c.params.ErrorRetryBaseDuration
-				} else {
-					interval = time.Duration(errorRetries) * time.Second
+			switch err := err.(type) {
+			case ExitReason:
+				// This is actually not an error case, but it causes an exit
+				c.recordSuccess()
+				c.lastError = err // This will be shown in the controller status
+				//goto shutdown
```

Comment:
No shutdown?

Reply:
oops, that was there to test my test... which apparently doesn't work. Thanks for catching it!

Reply:
fixed :)